<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Provides an R interface to the NCBIs EUtils API, 
    allowing users to search databases like GenBank 
    &lt;https://www.ncbi.nlm.nih.gov/genbank/&gt; and PubMed' 
    &lt;https://pubmed.ncbi.nlm.nih.gov/&gt;, process the 
    results of those searches and pull data into their R sessions.">
<title>Entrez in R • rentrez</title>
<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="/apple-touch-icon.png">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://docs.ropensci.org/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Entrez in R">
<meta property="og:description" content="Provides an R interface to the NCBIs EUtils API, 
    allowing users to search databases like GenBank 
    &lt;https://www.ncbi.nlm.nih.gov/genbank/&gt; and PubMed' 
    &lt;https://pubmed.ncbi.nlm.nih.gov/&gt;, process the 
    results of those searches and pull data into their R sessions.">
<meta property="og:image" content="https://docs.ropensci.org/rentrez/logo.png">
<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Matomo --><link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.css">
<script src="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.js" data-cfasync="false"></script><script src="https://ropensci.org/scripts/matomo.js"></script><noscript><p><img src="https://ropensci.matomo.cloud/matomo.php?idsite=1&amp;rec=1" style="border:0;" alt=""></p></noscript>
<!-- End Matomo Code -->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    <a href="https://ropensci.org" class="external-link"><img src="https://ropensci.org/img/icon_short_white.svg" id="hexlogo" alt="rOpenSci"></a>
    <a class="navbar-brand me-2" href="index.html">rentrez</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.2.3</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"></ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/rentrez_tutorial.html">Rentrez Tutorial</a>
  </div>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ropensci/rentrez/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9">
<div class="section level1">
<div class="page-header">
<img src="logo.png" class="logo" alt=""><h1 id="rentrez">rentrez<a class="anchor" aria-label="anchor" href="#rentrez"></a>
</h1>
</div>
<p><code>rentrez</code> provides functions that work with the <a href="http://www.ncbi.nlm.nih.gov/books/NBK25500/" class="external-link">NCBI Eutils</a> API to search, download data from, and otherwise interact with NCBI databases.</p>
<div class="section level2">
<h2 id="install">Install<a class="anchor" aria-label="anchor" href="#install"></a>
</h2>
<p><code>rentrez</code> is on CRAN, so you can get the latest stable release with <code>install.packages("rentrez")</code>. This repository will sometimes be a little ahead of the CRAN version, if you want the latest (and possibly greatest) version you can install the current github version using Hadley Wickham’s <a href="https://github.com/hadley/devtools" class="external-link">devtools</a>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(devtools)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install_github</span>(<span class="st">"ropensci/rentrez"</span>)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="the-eutils-api">The EUtils API<a class="anchor" aria-label="anchor" href="#the-eutils-api"></a>
</h2>
<p>Each of the functions exported by <code>rentrez</code> is documented, and this README and the package vignette provide examples of how to use the functions together as part of a workflow. The API itself is <a href="http://www.ncbi.nlm.nih.gov/books/NBK25500/" class="external-link">well-documented</a>. Be sure to read the official documentation to get the most out of API. In particular, be aware of the NCBI’s usage policies and try to limit very large requests to off peak (USA) times (<code>rentrez</code> takes care of limiting the number of requests per second, and setting the appropriate entrez tool name in each request).</p>
<p>Hopefully this README, and the package’s vignette and in-line documentation, provide you with enough information to get started with <code>rentrez</code>. If you need more help, or if you discover a bug in <code>rentrez</code> please let us know, either through one of the <a href="http://ropensci.org/contact.html" class="external-link">contact methods described here</a>, or <a href="https://github.com/ropensci/rentrez/issues" class="external-link">by filing an issue</a></p>
</div>
<div class="section level2">
<h2 id="examples">Examples<a class="anchor" aria-label="anchor" href="#examples"></a>
</h2>
<p>In many cases, doing something interesting with <code>EUtils</code> will take multiple calls. Here are a few examples of how the functions work together (check out the package vignette for others).</p>
<div class="section level3">
<h3 id="getting-data-from-that-great-paper-youve-just-read">Getting data from that great paper you’ve just read<a class="anchor" aria-label="anchor" href="#getting-data-from-that-great-paper-youve-just-read"></a>
</h3>
<p>Let’s say I’ve just read a paper on the evolution of Hox genes, <a href="dx.doi.org/10.1038/nature08789">Di-Poi <em>et al</em>. (2010)</a>, and I want to get the data required to replicate their results. First, I need the unique ID for this paper in pubmed (the PMID). Unfortunately, many journals don’t give PMIDS for their papers, but we can use <code>entrez_search</code> to find the paper using the doi field:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://docs.ropensci.org/rentrez/">rentrez</a></span><span class="op">)</span></span>
<span><span class="va">hox_paper</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/entrez_search.html">entrez_search</a></span><span class="op">(</span>db<span class="op">=</span><span class="st">"pubmed"</span>, term<span class="op">=</span><span class="st">"10.1038/nature08789[doi]"</span><span class="op">)</span></span>
<span><span class="va">hox_paper</span><span class="op">$</span><span class="va">ids</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] "20203609"</span></span></code></pre></div>
<p>Now, what sorts of data are available from other NCBI database for this paper?</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hox_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/entrez_link.html">entrez_link</a></span><span class="op">(</span>db<span class="op">=</span><span class="st">"all"</span>, id<span class="op">=</span><span class="va">hox_paper</span><span class="op">$</span><span class="va">ids</span>, dbfrom<span class="op">=</span><span class="st">"pubmed"</span><span class="op">)</span></span>
<span><span class="va">hox_data</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># elink object with contents:</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="co">#  $links: IDs for linked records from NCBI</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span></code></pre></div>
<p>In this case all the data is in the <code>links</code> element:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hox_data</span><span class="op">$</span><span class="va">links</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># elink result with information from 14 databases:</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="co">#  [1] pubmed_medgen              pubmed_pmc_refs           </span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="co">#  [3] pubmed_pubmed              pubmed_pubmed_alsoviewed  </span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="co">#  [5] pubmed_pubmed_citedin      pubmed_pubmed_combined    </span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="co">#  [7] pubmed_pubmed_five         pubmed_pubmed_reviews     </span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="co">#  [9] pubmed_pubmed_reviews_five pubmed_mesh_major         </span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="co"># [11] pubmed_nuccore             pubmed_nucleotide         </span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="co"># [13] pubmed_protein             pubmed_taxonomy_entrez</span></span></code></pre></div>
<p>Each of the character vectors in this object contain unique IDs for records in the named databases. These functions try to make the most useful bits of the returned files available to users, but they also return the original file in case you want to dive into the XML yourself.</p>
<p>In this case we’ll get the protein sequences as fasta files, using ’ <code>entrez_fetch</code>:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hox_proteins</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/entrez_fetch.html">entrez_fetch</a></span><span class="op">(</span>db<span class="op">=</span><span class="st">"protein"</span>, id<span class="op">=</span><span class="va">hox_data</span><span class="op">$</span><span class="va">links</span><span class="op">$</span><span class="va">pubmed_protein</span>, rettype<span class="op">=</span><span class="st">"fasta"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># No encoding supplied: defaulting to UTF-8.</span></span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/substr.html" class="external-link">substr</a></span><span class="op">(</span><span class="va">hox_proteins</span>, <span class="fl">1</span>, <span class="fl">237</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># &gt;gi|290760438|gb|ADD54588.1| HOXA10, partial [Saiphos equalis]</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="co"># MACSESPAANSFLVDSLISSASVRGEGGGGGGGGGGAGGGGGEGGGGGGGVYYPNNSSVYLPQTSELSYG</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="co"># LPSYGLFPVLSKRNEGPSQSMVPASHTYMSGMEVWLDPPRSCRLEDPESPQATSCSFTPNIKEENSYCLY</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="co"># DSDKGPKEATATDLSTFPRLTSEVCSMNNV</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="retrieving-datasets-associated-a-particular-organism">Retrieving datasets associated a particular organism.<a class="anchor" aria-label="anchor" href="#retrieving-datasets-associated-a-particular-organism"></a>
</h3>
<p>I like spiders. So let’s say I want to learn a little more about New Zealand’s endemic “black widow” the katipo. Specifically, in the past the katipo has been split into two species, can we make a phylogeny to test this idea?</p>
<p>The first step here is to use the function <code>entrez_search</code> to find datasets that include katipo sequences. The <code>popset</code> database has sequences arising from phylogenetic or population-level studies, so let’s start there.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://docs.ropensci.org/rentrez/">rentrez</a></span><span class="op">)</span></span>
<span><span class="va">katipo_search</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/entrez_search.html">entrez_search</a></span><span class="op">(</span>db<span class="op">=</span><span class="st">"popset"</span>, term<span class="op">=</span><span class="st">"Latrodectus katipo[Organism]"</span><span class="op">)</span></span>
<span><span class="va">katipo_search</span><span class="op">$</span><span class="va">count</span></span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] 6</span></span></code></pre></div>
<p>In this search <code>count</code> is the total number of hits returned for the search term. We can use <code>entrez_summary</code> to learn a little about these datasets. <code>rentrez</code> will parse this xml into a list of <code>esummary</code> records, with each list entry corresponding to one of the IDs it is passed. In this case we get six records, and we see what each one contains like so:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">katipo_summs</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/entrez_summary.html">entrez_summary</a></span><span class="op">(</span>db<span class="op">=</span><span class="st">"popset"</span>, id<span class="op">=</span><span class="va">katipo_search</span><span class="op">$</span><span class="va">ids</span><span class="op">)</span></span>
<span><span class="va">katipo_summs</span></span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># List of  6 esummary records. First record:</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="co">#  $`167843272`</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="co"># esummary result with 17 items:</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="co">#  [1] uid        caption    title      extra      gi         settype   </span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="co">#  [7] createdate updatedate flags      taxid      authors    article   </span></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a><span class="co"># [13] journal    strain     statistics properties oslt</span></span></code></pre></div>
<p>An we can extract specific elements from list of summary records with <code>extract_from_esummary</code>:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">titles</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/extract_from_esummary.html">extract_from_esummary</a></span><span class="op">(</span><span class="va">katipo_summs</span>, <span class="st">"title"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/unname.html" class="external-link">unname</a></span><span class="op">(</span><span class="va">titles</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] "Latrodectus katipo 18S ribosomal RNA gene, partial sequence; internal transcribed spacer 1, 5.8S ribosomal RNA gene, and internal transcribed spacer 2, complete sequence; and 28S ribosomal RNA gene, partial sequence."</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="co"># [2] "Latrodectus katipo cytochrome oxidase subunit 1 (COI) gene, partial cds; mitochondrial."                                                                                                                                 </span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="co"># [3] "Latrodectus 18S ribosomal RNA gene, partial sequence; internal transcribed spacer 1, 5.8S ribosomal RNA gene, and internal transcribed spacer 2, complete sequence; and 28S ribosomal RNA gene, partial sequence."       </span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="co"># [4] "Latrodectus cytochrome oxidase subunit 1 (COI) gene, partial cds; mitochondrial."                                                                                                                                        </span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="co"># [5] "Latrodectus tRNA-Leu (trnL) gene, partial sequence; and NADH dehydrogenase subunit 1 (ND1) gene, partial cds; mitochondrial."                                                                                            </span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a><span class="co"># [6] "Theridiidae cytochrome oxidase subunit I (COI) gene, partial cds; mitochondrial."</span></span></code></pre></div>
<p>Let’s just get the two mitochondrial loci (COI and trnL), using <code>entrez_fetch</code>:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">COI_ids</span> <span class="op">&lt;-</span> <span class="va">katipo_search</span><span class="op">$</span><span class="va">ids</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">6</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">trnL_ids</span> <span class="op">&lt;-</span> <span class="va">katipo_search</span><span class="op">$</span><span class="va">ids</span><span class="op">[</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="va">COI</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/entrez_fetch.html">entrez_fetch</a></span><span class="op">(</span>db<span class="op">=</span><span class="st">"popset"</span>, id<span class="op">=</span><span class="va">COI_ids</span>, rettype<span class="op">=</span><span class="st">"fasta"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># No encoding supplied: defaulting to UTF-8.</span></span></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">trnL</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/entrez_fetch.html">entrez_fetch</a></span><span class="op">(</span>db<span class="op">=</span><span class="st">"popset"</span>, id<span class="op">=</span><span class="va">trnL_ids</span>, rettype<span class="op">=</span><span class="st">"fasta"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># No encoding supplied: defaulting to UTF-8.</span></span></code></pre></div>
<p>The “fetched” results are fasta formatted characters, which can be written to disk easily:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/write.html" class="external-link">write</a></span><span class="op">(</span><span class="va">COI</span>, <span class="st">"Test/COI.fasta"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/write.html" class="external-link">write</a></span><span class="op">(</span><span class="va">trnL</span>, <span class="st">"Test/trnL.fasta"</span><span class="op">)</span></span></code></pre></div>
<p>Once you’ve got the sequences you can do what you want with them, but I wanted a phylogeny and we can do that entirly within R. To get a nice tree with legible tip labels I’m gong to use <code>stringr</code> to extract just the species names and <code>ape</code> to built and root and neighbor joining tree:</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://ape-package.ird.fr/" class="external-link">ape</a></span><span class="op">)</span></span>
<span><span class="va">tf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/write.html" class="external-link">write</a></span><span class="op">(</span><span class="va">COI</span>, <span class="va">tf</span><span class="op">)</span></span>
<span><span class="va">coi</span> <span class="op">&lt;-</span> <span class="fu">read.dna</span><span class="op">(</span><span class="va">tf</span>, format<span class="op">=</span><span class="st">"fasta"</span><span class="op">)</span></span>
<span><span class="va">coi_aligned</span> <span class="op">&lt;-</span> <span class="fu">muscle</span><span class="op">(</span><span class="va">coi</span><span class="op">)</span></span>
<span><span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu">nj</span><span class="op">(</span><span class="fu">dist.dna</span><span class="op">(</span><span class="va">coi_aligned</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tree</span><span class="op">$</span><span class="va">tip.label</span> <span class="op">&lt;-</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_extract.html" class="external-link">str_extract</a></span><span class="op">(</span><span class="va">tree</span><span class="op">$</span><span class="va">tip.label</span>, <span class="st">"Steatoda [a-z]+|Latrodectus [a-z]+"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span> <span class="fu">root</span><span class="op">(</span><span class="va">tree</span>, outgroup<span class="op">=</span><span class="st">"Steatoda grossa"</span> <span class="op">)</span>, cex<span class="op">=</span><span class="fl">0.8</span><span class="op">)</span></span></code></pre></div>
<p><img src="http://i.imgur.com/8n9UeIi.png"></p>
</div>
<div class="section level3">
<h3 id="web_history-and-big-queries">web_history and big queries<a class="anchor" aria-label="anchor" href="#web_history-and-big-queries"></a>
</h3>
<p>The NCBI provides search history features, which can be useful for dealing with large lists of IDs or repeated searches.</p>
<p>As an example, imagine you wanted to learn something about all of the SNPs in the non-recombing portion of the Y chromsome in humans. You could first find these SNPs using <code>entrez_search</code>, using the “CHR” (chromosome) and “CPOS” (position in chromosome) to specify the region of interest. (The syntax for these search terms is described in the vignette and the documentation for <code>entrez_search</code>):</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">snp_search</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/entrez_search.html">entrez_search</a></span><span class="op">(</span>db<span class="op">=</span><span class="st">"snp"</span>, </span>
<span>                            term<span class="op">=</span><span class="st">"(Y[CHR] AND Homo[ORGN]) NOT 10001:2781479[CPOS]"</span><span class="op">)</span></span>
<span><span class="va">snp_search</span></span></code></pre></div>
<div class="sourceCode" id="cb25"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Entrez search result with 234154 hits (object contains 20 IDs and no web_history object)</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="co">#  Search term (as translated):  (Y[CHR] AND "Homo"[Organism]) NOT 10001[CHRPOS] :  ...</span></span></code></pre></div>
<p>When I wrote this that was a little over 200 000 SNPs. It’s probably not a good idea to set <code>retmax</code> to 200 000 and just download all of those identifiers. Instead, we could store this list of IDs on the NCBI’s server and refer to them in later calles to functions like <code>entrez_link</code> and <code>entrez_fetch</code> that accept a web history object.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">snp_search</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/entrez_search.html">entrez_search</a></span><span class="op">(</span>db<span class="op">=</span><span class="st">"snp"</span>, </span>
<span>                            term<span class="op">=</span><span class="st">"(Y[CHR] AND Homo[ORGN]) NOT 10001:2781479[CPOS]"</span>, </span>
<span>                            use_history <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">snp_search</span></span></code></pre></div>
<div class="sourceCode" id="cb27"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Entrez search result with 234154 hits (object contains 20 IDs and a web_history object)</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="co">#  Search term (as translated):  (Y[CHR] AND "Homo"[Organism]) NOT 10001[CHRPOS] :  ...</span></span></code></pre></div>
<p>As you can see, the result of the search now includes a <code>web_history</code> object. We can use that object to refer to these IDs in later calls. Heree we will just fetch complete records of the first 5 SNPs.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">recs</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/entrez_fetch.html">entrez_fetch</a></span><span class="op">(</span>db<span class="op">=</span><span class="st">"snp"</span>, web_history<span class="op">=</span><span class="va">snp_search</span><span class="op">$</span><span class="va">web_history</span>, retmax<span class="op">=</span><span class="fl">5</span>, rettype<span class="op">=</span><span class="st">"xml"</span>, parsed<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">recs</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb29"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] "XMLInternalDocument" "XMLAbstractDocument"</span></span></code></pre></div>
<p>The records come to us as parsed XML objects, which you could futher process with the <code>XML</code> library or write to disk for later use.</p>
</div>
<div class="section level3">
<h3 id="getting-information-about-ncbi-databases">Getting information about NCBI databases<a class="anchor" aria-label="anchor" href="#getting-information-about-ncbi-databases"></a>
</h3>
<p>Most of the examples above required some background information about what databases NCBI has to offer, and how they can be searched. <code>rentrez</code> provides a set of functions with names starting <code>entrez_db</code> that help you to discover this information in an interactive session.</p>
<p>First up, <code><a href="reference/entrez_dbs.html">entrez_dbs()</a></code> gives you a list of database names</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/entrez_dbs.html">entrez_dbs</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb31"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co">#  [1] "pubmed"          "protein"         "nuccore"        </span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="co">#  [4] "nucleotide"      "nucgss"          "nucest"         </span></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a><span class="co">#  [7] "structure"       "genome"          "annotinfo"      </span></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a><span class="co"># [10] "assembly"        "bioproject"      "biosample"      </span></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a><span class="co"># [13] "blastdbinfo"     "books"           "cdd"            </span></span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a><span class="co"># [16] "clinvar"         "clone"           "gap"            </span></span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a><span class="co"># [19] "gapplus"         "grasp"           "dbvar"          </span></span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a><span class="co"># [22] "epigenomics"     "gene"            "gds"            </span></span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a><span class="co"># [25] "geoprofiles"     "homologene"      "medgen"         </span></span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a><span class="co"># [28] "mesh"            "ncbisearch"      "nlmcatalog"     </span></span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a><span class="co"># [31] "omim"            "orgtrack"        "pmc"            </span></span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a><span class="co"># [34] "popset"          "probe"           "proteinclusters"</span></span>
<span id="cb31-13"><a href="#cb31-13" aria-hidden="true" tabindex="-1"></a><span class="co"># [37] "pcassay"         "biosystems"      "pccompound"     </span></span>
<span id="cb31-14"><a href="#cb31-14" aria-hidden="true" tabindex="-1"></a><span class="co"># [40] "pcsubstance"     "pubmedhealth"    "seqannot"       </span></span>
<span id="cb31-15"><a href="#cb31-15" aria-hidden="true" tabindex="-1"></a><span class="co"># [43] "snp"             "sra"             "taxonomy"       </span></span>
<span id="cb31-16"><a href="#cb31-16" aria-hidden="true" tabindex="-1"></a><span class="co"># [46] "unigene"         "gencoll"         "gtr"</span></span></code></pre></div>
<p>Some of the names are a little opaque, so you can get some more descriptive information about each with <code><a href="reference/entrez_db_summary.html">entrez_db_summary()</a></code></p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/entrez_db_summary.html">entrez_db_summary</a></span><span class="op">(</span><span class="st">"cdd"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb33"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co">#  DbName: cdd</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="co">#  MenuName: Conserved Domains</span></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a><span class="co">#  Description: Conserved Domain Database</span></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a><span class="co">#  DbBuild: Build150814-1106.1</span></span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a><span class="co">#  Count: 50648</span></span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a><span class="co">#  LastUpdate: 2015/08/14 18:42</span></span></code></pre></div>
<p><code><a href="reference/entrez_db_searchable.html">entrez_db_searchable()</a></code> lets you discover the fields available for search terms for a given database. You get back a named-list, with names are fields. Each element has additional information about each named search field (you can also use <code>as.data.frame</code> to create a dataframe, with one search-field per row):</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">search_fields</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/entrez_db_searchable.html">entrez_db_searchable</a></span><span class="op">(</span><span class="st">"pmc"</span><span class="op">)</span></span>
<span><span class="va">search_fields</span><span class="op">$</span><span class="va">GRNT</span></span></code></pre></div>
<div class="sourceCode" id="cb35"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co">#  Name: GRNT</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="co">#  FullName: Grant Number</span></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a><span class="co">#  Description: NIH Grant Numbers</span></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a><span class="co">#  TermCount: 2272841</span></span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a><span class="co">#  IsDate: N</span></span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a><span class="co">#  IsNumerical: N</span></span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a><span class="co">#  SingleToken: Y</span></span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a><span class="co">#  Hierarchy: N</span></span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a><span class="co">#  IsHidden: N</span></span></code></pre></div>
<p>Finally, <code>entrez_db_links</code> takes a database name, and returns a list of other NCBI databases which might contain linked-records.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/entrez_db_links.html">entrez_db_links</a></span><span class="op">(</span><span class="st">"omim"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb37"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Databases with linked records for database 'omim'</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a><span class="co">#  [1] biosample   biosystems  books       clinvar     dbvar      </span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a><span class="co">#  [6] gene        genetests   geoprofiles gtr         homologene </span></span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a><span class="co"># [11] mapview     medgen      medgen      nuccore     nucest     </span></span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a><span class="co"># [16] nucgss      omim        pcassay     pccompound  pcsubstance</span></span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a><span class="co"># [21] pmc         protein     pubmed      pubmed      sra        </span></span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a><span class="co"># [26] structure   unigene</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="trendy-topics-in-genetics">Trendy topics in genetics<a class="anchor" aria-label="anchor" href="#trendy-topics-in-genetics"></a>
</h3>
<p>This is one is a little more trivial, but you can also use entrez to search pubmed and the EUtils API allows you to limit searches by the year in which the paper was published. That gives is a chance to find the trendiest -omics going around (this has quite a lot of repeated searching, so it you want to run your own version be sure to do it in off peak times).</p>
<p>Let’s start by making a function that finds the number of records matching a given search term for each of several years (using the <code>mindate</code> and <code>maxdate</code> terms from the Eutils API):</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rentrez)</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>papers_by_year <span class="ot">&lt;-</span> <span class="cf">function</span>(years, search_term){</span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">sapply</span>(years, <span class="cf">function</span>(y) <span class="fu">entrez_search</span>(<span class="at">db=</span><span class="st">"pubmed"</span>,<span class="at">term=</span>search_term, <span class="at">mindate=</span>y, <span class="at">maxdate=</span>y, <span class="at">retmax=</span><span class="dv">0</span>)<span class="sc">$</span>count))</span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p>With that we can fetch the data for each term and, by searching with no term, find the total number of papers published in each year:</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">years</span> <span class="op">&lt;-</span> <span class="fl">1990</span><span class="op">:</span><span class="fl">2015</span></span>
<span><span class="va">total_papers</span> <span class="op">&lt;-</span> <span class="fu">papers_by_year</span><span class="op">(</span><span class="va">years</span>, <span class="st">""</span><span class="op">)</span></span>
<span><span class="va">omics</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"genomic"</span>, <span class="st">"epigenomic"</span>, <span class="st">"metagenomic"</span>, <span class="st">"proteomic"</span>, <span class="st">"transcriptomic"</span>, <span class="st">"pharmacogenomic"</span>, <span class="st">"connectomic"</span> <span class="op">)</span></span>
<span><span class="va">trend_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">omics</span>, <span class="kw">function</span><span class="op">(</span><span class="va">t</span><span class="op">)</span> <span class="fu">papers_by_year</span><span class="op">(</span><span class="va">years</span>, <span class="va">t</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">trend_props</span> <span class="op">&lt;-</span> <span class="va">trend_data</span><span class="op">/</span><span class="va">total_papers</span></span></code></pre></div>
<p>That’s the data, let’s plot it:</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://had.co.nz/reshape" class="external-link">reshape</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="va">trend_df</span> <span class="op">&lt;-</span> <span class="fu">melt</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">years</span>, <span class="va">trend_props</span><span class="op">)</span>, id.vars<span class="op">=</span><span class="st">"years"</span><span class="op">)</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">trend_df</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">years</span>, <span class="va">value</span>, colour<span class="op">=</span><span class="va">variable</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">p</span> <span class="op">+</span> <span class="fu">geom_line</span><span class="op">(</span>size<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_y_log10</span><span class="op">(</span><span class="st">"number of papers"</span><span class="op">)</span></span></code></pre></div>
<p>Giving us… well this:</p>
<p><img src="http://i.imgur.com/oSYuWqz.png"></p>
<hr>
<p>This package is part of a richer suite called <a href="https://github.com/ropensci/fulltext" class="external-link">fulltext</a>, along with several other packages, that provides the ability to search for and retrieve full text of open access scholarly articles.</p>
<hr>
</div>
</div>
</div>

  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://cloud.r-project.org/package=rentrez" class="external-link">View on CRAN</a></li>
<li><a href="https://github.com/ropensci/rentrez/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/ropensci/rentrez/issues/" class="external-link">Report a bug</a></li>
</ul>
</div>



<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>

<div class="community">
<h2 data-toc-skip>Community</h2>
<ul class="list-unstyled">
<li><a href="CONTRIBUTING.html">Contributing guide</a></li>
</ul>
</div>

<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing rentrez</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>David Winter <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-6165-0029" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>

<div class="r-universe">
<h2 data-toc-skip>R-universe</h2>
<ul class="list-unstyled">
<li><p><a href="https://ropensci.r-universe.dev/rentrez" class="external-link"><img src="https://ropensci.r-universe.dev/badges/rentrez" alt="rentrez status badge"></a></p></li>
</ul>
</div>

  </aside>
</div>


    <footer><!-- begin footer --><div class="footer">
    <div class="container">
        <div class="row start top-4 bottom-8">
            <div class="col-2"> <img id="footerlogo" src="https://ropensci.org/img/icon_short_white.svg">
</div>
            <div class="col-10">
                <div class="row">
                    <div class="col-md-4 col-xs-6">
                        <a href="https://github.com/ropensci" target="_blank" class="external-link"><div class="icon fab fa-github"></div></a>
                        <a href="https://github.com/ropenscilabs" target="_blank" class="external-link"><div class="icon fa fa-flask"></div></a>
                        <a href="https://hachyderm.io/@ropensci" target="_blank" class="external-link"><div class="icon fab fa-mastodon"></div></a>
                        <a href="https://vimeo.com/ropensci" target="_blank" class="external-link"><div class="icon fab fa-vimeo"></div></a>
                    </div>
                </div>
                <div class="row top-4">
                    <div class="col-md-2 col-sm-4">
                        <ul>
<h5 class="bottom-2">About</h5>
                            <li><a href="https://ropensci.org/about" class="external-link">About rOpenSci</a></li>
                            <li><a href="https://ropensci.org/software-review" class="external-link">Software Review</a></li>
                            <li><a href="https://ropensci.org/about#team" class="external-link">Our Team</a></li>
                            <li><a href="https://ropensci.org/careers" class="external-link">Jobs</a></li>
                            <li><a href="https://ropensci.org/donate" class="external-link">Donate</a></li>
                            <li><a href="https://ropensci.org/contact" class="external-link">Contact Us</a></li>
                        </ul>
</div>
                    <div class="col-md-3 col-sm-4">
                        <ul>
<h5 class="bottom-2">Community</h5>
                            <li><a href="https://ropensci.org/community/" class="external-link">Our Community</a></li>
                            <li><a href="https://ropensci.org/commcalls/" class="external-link">Community calls</a></li>
                            <li><a href="https://ropensci.org/events/" class="external-link">Events</a></li>
                            <li><a href="https://discuss.ropensci.org/" class="external-link">Join the Discussion</a></li>
                            <li><a href="https://ropensci.org/code-of-conduct" class="external-link">Code of conduct</a></li>
                        </ul>
</div>
                    <div class="col-md-2 col-sm-4">
                        <ul>
<h5 class="bottom-2">Resources</h5>
                            <li><a href="https://ropensci.org/packages/" class="external-link">Packages</a></li>
                            <li><a href="https://ropensci.org/usecases/" class="external-link">Use Cases</a></li>
                            <li><a href="https://ropensci.org/talks-papers/" class="external-link">Talks &amp; Publications</a></li>
                            <li><a href="https://docs.ropensci.org/" class="external-link">Documentation</a></li>
                            <li><a href="https://ropensci.org/news/" class="external-link">Newsletter</a></li>
                            <li><a href="https://ropensci.org/how-to-cite-ropensci/" class="external-link">Cite rOpenSci</a></li>
                        </ul>
</div>
                    <div class="col-md-4 col-xs-12">
                        <h5 class="bottom-2"></h5>

                        <p>rOpenSci is a fiscally sponsored project of <a href="http://numfocus.org" class="external-link">NumFOCUS</a>.</p>

                    </div>
                </div>
            </div>
        </div>
    </div>
</div>
<!-- / end footer -->


    </footer>
</div>

  

  

  </body>
</html>
